src/app/components/lineage-graph/lineage-graph.component.ts

```diff
@@ -23,10 +20,10 @@ import { LineageGraphNodeKind } from "src/app/dataset-view/additional-components
     styleUrls: ["./lineage-graph.component.scss"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class LineageGraphComponent implements OnChanges, OnInit {
-    @Input() public links: Edge[];
-    @Input() public nodes: Node[];
-    @Input() public currentDataset: DatasetLineageBasicsFragment;
+export class LineageGraphComponent {
+    @Input() public links: MaybeNull<Edge[]>;
+    @Input() public nodes: MaybeNull<Node[]>;
+    @Input() public currentDataset: MaybeNull<DatasetLineageBasicsFragment>;
```

Comment:
Can you explain when it can be null?
I think this situation is impossible in the way we organized the `DatasetComonent`. The graph component is always under `ngIf` that checks the model is present.

Reply:
Ok, removed MaybeNull